tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -561,3 +562,190 @@ def test_invalid(self):
+    # adapted from scipy's analogous tests
+    def _inner_cases(self, original, xp, sp, dtype):
+        if(self.inner_modification == 'normal'):
+            return (xp.array(original, dtype=dtype), original)
+        if(self.inner_modification == 'sparse'):
+            return (sp.csr_matrix(
+                    original, dtype=dtype), original)
+
+        # creating base-matrix-like class with default
+        # matrix-vector and adjoint-matrix-vector impl
+        def mv(x, dtype):
+            y = original.dot(x)
+            if len(x.shape) == 2:
+                y = y.reshape(-1, 1)
+            return y
+
+        def rmv(x, dtype):
+            return original.T.conj().dot(x)
+
+        # defining the base-class
+        class BaseMatlike(sp.linalg.LinearOperator):
+            args = ()
+
+            def __init__(self, dtype):
+                self.dtype = xp.dtype(dtype)
+                self.shape = original.shape
+
+            def _matvec(self, x):
+                return mv(x, self.dtype)
+
+        # using above base-class to make and test sub-classes based
+        # on the current `inner_modification` parameter
+        if(self.inner_modification == 'adjoint'):
+
+            # defining the sub-class
+            class HasAdjoint(BaseMatlike):
+                args = ()
+
+                def _adjoint(self):
+                    shape = self.shape[1], self.shape[0]
+                    matvec = partial(rmv, dtype=self.dtype)
+                    rmatvec = partial(mv, dtype=self.dtype)
+                    return sp.linalg.LinearOperator(matvec=matvec,
+                                                    rmatvec=rmatvec,
+                                                    dtype=self.dtype,
+                                                    shape=shape)
+            return (HasAdjoint(dtype), original)
+
+        # defining the sub-classes
+        class HasRmatvec(BaseMatlike):
+            args = ()
+
+            def _rmatvec(self, x):
+                return rmv(x, self.dtype)
+
+        class HasRmatmat(HasRmatvec):
+
+            def _matmat(self, x):
+                return original.dot(x)
+
+            def _rmatmat(self, x):
+                return original.T.conj().dot(x)
+
+        if(self.inner_modification == 'adjoint-matrix-vector'):
+            return (HasRmatvec(dtype), original)
+        if(self.inner_modification == 'adjoint-matrix-matrix'):
+            return (HasRmatmat(dtype), original)
+
+    def _outer_cases(self, xp, sp):
+        if(self.dtype != xp.complex_):
+            original = xp.array([[1., 2., 3.], [4., 5., 6.]])
+        else:
+            original = xp.array([[1, 2j, 3j], [4j, 5j, 6]])
+
+        if(self.outer_modification == 'normal'):
+            return self._inner_cases(original, xp, sp, self.dtype)
+        if(self.outer_modification == 'transpose'):
+            M, A = self._inner_cases(original.T, xp, sp, self.dtype)
+            return (sp.linalg.aslinearoperator(M).T, A.T)
```

Comment:
If the first element of the return value is always expected to be given as the argument of `linearoperator`, How about transforming here?

Reply:
I'm afraid I don't understand @asi1024 , what do you mean by 'transforming' here?